### ISSUE TITLE
Macro not recording auto-completion result

### ISSUE THREAD

**Issue by yishayw** [vrapper/vrapper#791]
Steps to reproduce:

- Start recording macro (qq)
- Use auto completing (ctrl-space +  <some-keys> arrow keys + enter)
- Stop recording (q)
- Replay macro

Desired result: the result of pressing <enter> will appear in the played macro result.
Actual result: Only <some-keys> appear, so I have to manually complete on each macro replay

**Comment by albertdev** [comment]
Thank you for letting us know. 

I can reproduce that recording a macro and doing command completion in Insert mode does not record the final text in the macro.

However, is this exactly what you are talking about? Just to be clear, while the "repeat insert" works (i.e. pressing <kbd>.</kbd>), Vrapper never records that you start command completion and select the second item. Instead the repeat will simply insert the same text as edited the last time Insert mode was used.

In other words, the fix I'm thinking of will simply record the text result from the last insert session but it won't do a real command completion.

Please let us know.

**Comment by yishayw** [comment]
That fix was what I had in mind. Great to have such a swift response. Thanks.

**Comment by keforbes** [comment]
@albertdev, how would you capture the final text?  Would you check to see if we're recording when we exit insert mode and if so, capture whatever was entered?

For what it's worth, 5 years ago [I mentioned](https://github.com/vrapper/vrapper/issues/133#issuecomment-10875713) an issue where using the arrow keys while in insert mode wouldn't be captured by a macro.  That issue is now fixed.  Would we be able to ensure that non-text key presses (arrow keys) while in insert mode would still be captured in the macro if we're just capturing the final text entered?

**Comment by albertdev** [comment]
@keyforbes, that is a valid remark. Since I was thinking of just capturing the text when we exit Insert mode that arrow-key-recording feature would no longer work.

Of course, if this new behavior can be enabled or disabled by an option then you can always choose to toggle it when you record a macro or set the desired default behavior from your `.vrapperrc`.

Your remark shows though that we can't make it the default as that would surprise our existing users or new people who want a Vim-like experience.
